[ROOT626_X] Compilation failure in PhysicsTools/TagAndProbe

Error message:
```
>> Compiling  src/PhysicsTools/TagAndProbe/src/ZGeneratorLineShape.cc
/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_14_1_ROOT626_X_2024-03-04-2300' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_14_1_ROOT626_X_2024-03-04-2300' -I/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/fe9231d96c033d124d91db8e46c74763/opt/cmssw/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_1_ROOT626_X_2024-03-04-2300/src -I/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/fe9231d96c033d124d91db8e46c74763/opt/cmssw/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_1_ROOT626_X_2024-03-04-2300/poison -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_ROOT626_X_2024-03-03-0000/src -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/cms/coral/CORAL_2_3_21-1d266413eb10ae5589055293c0a5ffd1/include/LCG -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/alpaka/1.1.0-0e0b978d445f7af747cf00064c146356/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/pcre/8.43-37eb2e8b73bab83d6645ecfd5d73dcaa/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/boost/1.80.0-9d4e0dc8cf1f8fc8a67cf53ae917955e/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/clhep/2.4.7.1-5c1cfd46e8f0e61f022ff44f21c90ce1/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/curl/7.79.0-959a9188e043d67b2825f64cfeb54266/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/gsl/2.6-dc093739c7d4283cbe37712ffdbef83e/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/hepmc/2.06.10-84d62bf8401502062c47cad693db65fe/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/hls/2019.08-8afb4083e7b06154cf0bca6d787b688f/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/hls4mlEmulatorExtras/1.1.1-c788aa889fa249afee5adfca90f0c6af/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/lcg/root/6.26.15-c3ab3debfabae9b54d772fef50482897/include -isystem/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/tbb/v2021.9.0-5109e373398eaf79a8268d6a86a2f6e2/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/utm/utm_0.12.0-c3920c78ffe53731d597ca22bdbf0646/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/cms/vdt/0.4.3-5a80085534117eaccb28e669c6da4b6f/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/xerces-c/3.1.3-c7b88eaa36d0408120f3c29826a04bf6/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/zlib/1.2.11-51072030b7f93c3ac6c4235f21e413cb/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/eigen/3bb6a48d8c171cf20b5f8e48bfb4e424fbd4f79e-cf66836cbd6ad8d696d804181e36d61c/include/eigen3 -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/fmt/8.0.1-54e94b39f5cf29341bb9c4765764e1ca/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/hepmc3/3.2.7-fd16dc6ff1399275c883ed21e455694f/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/OpenBLAS/0.3.15-15504a5c800a9ecbbc2befbb991bead5/include -I/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/external/tinyxml2/6.2.0-d17873b4d6a42a43226cf689f82ec1ef/include -O2 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++17 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -msse3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-deprecated-copy -Wno-deprecated -Wno-error=unused-variable -DALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT=128 -DALPAKA_DISABLE_VENDOR_RNG -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/PhysicsTools/TagAndProbe/src/PhysicsToolsTagAndProbe/ZGeneratorLineShape.cc.d src/PhysicsTools/TagAndProbe/src/ZGeneratorLineShape.cc -o tmp/el8_amd64_gcc12/src/PhysicsTools/TagAndProbe/src/PhysicsToolsTagAndProbe/ZGeneratorLineShape.cc.o
src/PhysicsTools/TagAndProbe/src/TagProbeFitter.cc: In member function 'void TagProbeFitter::doFitEfficiency(RooWorkspace*, std::string, RooRealVar&)':
  src/PhysicsTools/TagAndProbe/src/TagProbeFitter.cc:561:15: error: cannot bind non-const lvalue reference of type 'RooAbsData&' to an rvalue of type 'RooAbsData'
   561 |     w->import(RooDataHist{"data_binned", "data_binned", *obs, *data});
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/fe9231d96c033d124d91db8e46c74763/opt/cmssw/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_1_ROOT626_X_2024-03-04-2300/src/PhysicsTools/TagAndProbe/interface/TagProbeFitter.h:7,
                 from src/PhysicsTools/TagAndProbe/src/TagProbeFitter.cc:1:
/data/cmsbld/jenkins/workspace/build-any-ib/w/el8_amd64_gcc12/lcg/root/6.26.15-c3ab3debfabae9b54d772fef50482897/include/RooWorkspace.h:66:29: note:   initializing argument 1 of 'Bool_t RooWorkspace::import(RooAbsData&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&, const RooCmdArg&)'
   66 |   Bool_t import(RooAbsData& data,
      |                 ~~~~~~~~~~~~^~~~
  gmake: *** [tmp/el8_amd64_gcc12/src/PhysicsTools/TagAndProbe/src/PhysicsToolsTagAndProbe/TagProbeFitter.cc.o] Error 1
```

Full log: [link](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/el8_amd64_gcc12/CMSSW_14_1_ROOT626_X_2024-03-04-2300/PhysicsTools/TagAndProbe)

Most likely caused by #44069 - @guitargeek

--- a/PhysicsTools/TagAndProbe/src/TagProbeFitter.cc
+++ b/PhysicsTools/TagAndProbe/src/TagProbeFitter.cc
@@ -558,7 +558,8 @@ void TagProbeFitter::doFitEfficiency(RooWorkspace* w, string pdfName, RooRealVar
     const RooArgSet* dataObs = data->get(0);
     // remove everything which is not a dependency of the pdf
     std::unique_ptr<RooArgSet> obs{w->pdf("simPdf")->getObservables(dataObs)};
-    w->import(RooDataHist{"data_binned", "data_binned", *obs, *data});
+    RooDataHist bdata{"data_binned", "data_binned", *obs, *data};
+    w->import(bdata);
     data = w->data("data_binned");
   }
 

